tsl/src/remote/txn_resolve.c

```diff
@@ -22,10 +22,21 @@ remote_txn_resolution(Oid foreign_server, const RemoteTxnId *transaction_id)
 		/* transaction still ongoing; don't know it's state */
 		return REMOTE_TXN_RESOLUTION_UNKNOWN;
 
+	/*
+	 * If an entry exists in the "remote_txn" table and is visible then it means
+	 * that the transaction committed on the AN
+	 */
 	if (remote_txn_persistent_record_exists(transaction_id))
 		return REMOTE_TXN_RESOLUTION_COMMT;
 
-	return REMOTE_TXN_RESOLUTION_ABORT;
+	/*
+	 * We need to check the Postgres machinery to ensure that this txn has indeed
+	 * aborted before declaring so.
+	 */
+	if (TransactionIdDidAbort(transaction_id->xid))
+		return REMOTE_TXN_RESOLUTION_ABORT;
+	else /* something's still going on with this transaction */
```

Comment:
Do we know some scenario where this happens? It could be helpful to describe it in this comment.

Reply:
It is a concurrent and still ongoing transaction where there is no decisions on COMMIT or ABORT yet. Thus no healing or cleanup decision can be made.

I agree that the comment can be more explanatory though.